UI/obs-app.cpp

```diff
@@ -734,120 +689,211 @@ bool OBSApp::UpdatePre22MultiviewLayout(const char *layout)
 bool OBSApp::InitGlobalConfig()
 {
 	char path[512];
-	bool changed = false;
 
-	int len = GetConfigPath(path, sizeof(path), "obs-studio/global.ini");
+	int len = GetAppConfigPath(path, sizeof(path), "obs-studio/global.ini");
 	if (len <= 0) {
 		return false;
 	}
 
-	int errorcode = globalConfig.Open(path, CONFIG_OPEN_ALWAYS);
+	int errorcode = appConfig.Open(path, CONFIG_OPEN_ALWAYS);
 	if (errorcode != CONFIG_SUCCESS) {
 		OBSErrorBox(NULL, "Failed to open global.ini: %d", errorcode);
 		return false;
 	}
 
+	uint32_t lastVersion =
+		config_get_int(appConfig, "General", "LastVersion");
+
+	bool migratedUserSettings =
+		config_get_bool(appConfig, "General", "UserSettingsMigrated");
+
+	if (!migratedUserSettings &&
+	    lastVersion < MAKE_SEMANTIC_VERSION(31, 0, 0)) {
+		bool migrated = MigrateGlobalSettings();
+
+		config_set_bool(appConfig, "General", "UserSettingsMigrated",
+				migrated);
```

Comment:
The idea is that the "global" settings have been migrated _to_ the "user" settings, which should happen if the last version OBS ran as is lower than whichever version we release this change with (which I'd also need to adapt accordingly).

I might need to give this another pass anyway, because I don't think the migration setting is set on a "fresh" OBS install (as there is nothing to migrate there anyway), but the key idea here is to mark the case of "legacy global config exists and was already successfully migrated".

Reply:
I've changed this code now a bit - I call the setting `Pre31Migrated` to keep it in line with the other similar settings.

I also check for the older version first (I'd still need to change this to the actual version we target for this change) and for the migration flag second.

This mainly serves as a guard against unnecessary duplicate migrations which will trigger an error message as the copied `user.ini` already exists.

Reply:
Changes look fine at a glance.